utils/obsproc/NetCDFToIodaConverter.h

```diff
@@ -93,6 +93,52 @@ namespace gdasapp {
       location += other.location;
       oops::Log::trace() << "IodaVars::IodaVars done appending." << std::endl;
     }
+    void trim(const Eigen::Array<bool, Eigen::Dynamic, 1>& mask ) {
+      int newlocation = mask.count();
+
+      Eigen::ArrayXf longitudeMasked(newlocation);
+      Eigen::ArrayXf latitudeMasked(newlocation);
+      Eigen::Array<int64_t, Eigen::Dynamic, 1> datetimeMasked(newlocation);
+      Eigen::ArrayXf obsValMasked(newlocation);
+      Eigen::ArrayXf obsErrorMasked(newlocation);
+      Eigen::ArrayXi preQcMasked(newlocation);
+
+      Eigen::ArrayXXf floatMetadataMasked(newlocation, nfMetadata);
+      Eigen::ArrayXXf intMetadataMasked(newlocation, niMetadata);
+
+      // this feels downright crude, but apparently numpy-type masks are on the todo list for Eigen
+      int j = 0;
+      for (int i = 0; i < location; i++) {
+        if (mask(i)) {
+          longitudeMasked(j) = longitude(i);
+          latitudeMasked(j) = latitude(i);
+          obsValMasked(j) = obsVal(i);
+          obsErrorMasked(j) = obsError(i);
+          preQcMasked(j) = preQc(i);
+          datetimeMasked(j) = datetime(i);
+          for (int k = 0; k < nfMetadata; k++) {
+            intMetadataMasked(j, k) = intMetadata(i, k);
+            }
+          for (int k = 0; k < niMetadata; k++) {
+            intMetadataMasked(j, k) = intMetadata(i, k);
+            }
+        j++;
+        }  // end if (mask(i))
+      }
+
+      longitude = longitudeMasked;
```

Comment:
**nitpick**: should class variables end in underscores to indicate their status (ex: `longitude_`). Do we have a convention for this?

Reply:
Yes @rmclaren ... my bad. I started doing this for the other class but I forgot to do it for IodaVars ...

Reply:
So I presume this needs to be changed in IodaVars and any currently merged converters, all at once? I.e., not in this PR?